OmniKitUI/ViewControllers/ReplacePodViewController.swift

```diff
@@ -25,19 +25,23 @@ class ReplacePodViewController: SetupTableViewController {
-                case .occluded,                    // $14
-                     .occlusionCheckValueTooHigh, .occlusionCheckStartup1, .occlusionCheckStartup2,         // $5A, $60, $60
-                     .occlusionCheckTimeouts1, .occlusionCheckTimeouts2, .occlusionCheckTimeouts3,          // $62, $66, $67
-                     .occlusionCheckPulseIssue, .occlusionCheckBolusProblem, .occlusionCheckAboveThreshold: // $68, $69, $6A
-                    instructionsLabel.text = LocalizedString("Occlusion detected. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an occlusion")
-                default:
-                    instructionsLabel.text = String(format: LocalizedString("The pod has detected internal fault 0x%02x. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to a fault (1: The fault code value)"), eventType.rawValue)
+            case .fault(let faultCode):
+                if let faultType = faultCode.faultType {
+                    switch faultType {
+                    case .reservoirEmpty:              // $18
+                       instructionsLabel.text = LocalizedString("Empty reservoir. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an empty reservoir")
+                    case .exceededMaximumPodLife80Hrs: // $1C
+                       instructionsLabel.text = LocalizedString("Pod expired. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an expired pod")
+                    case .occluded,                    // $14
+                         .occlusionCheckValueTooHigh, .occlusionCheckStartup1, .occlusionCheckStartup2,         // $5A, $60, $60
+                         .occlusionCheckTimeouts1, .occlusionCheckTimeouts2, .occlusionCheckTimeouts3,          // $62, $66, $67
+                         .occlusionCheckPulseIssue, .occlusionCheckBolusProblem, .occlusionCheckAboveThreshold: // $68, $69, $6A
+                        instructionsLabel.text = LocalizedString("Occlusion detected. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an occlusion")
+                    default:
+                        instructionsLabel.text = String(format: LocalizedString("The pod has detected internal fault %d. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to a fault (1: The fault code value)"), faultType.rawValue)
+                    }
+                } else {
+                    instructionsLabel.text = LocalizedString("The pod has detected an internal fault. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to a fault ")
```

Comment:
You can use `faultCode.rawValue` here to show the actual fault code number.

Reply:
If I try to use faultCode.rawValue instead of faultType.rawValue to show the actual fault code number on line 41, I get a Swift Compile Error of "‘rawValue’ is inaccessible due to ‘internal’ protection level". The current code compiles cleanly and handles all the simulated fault conditions I injected correctly.

Reply:
Ah, yes, you'll need to expose rawType on FaultEventCode as public. This is the only way to access codes like 0x04, or 0x0C if we ever happened to receive them from a pod.

Reply:
Yes. This has been the underlying issue all along and why I originally was mapping unknown fault errors to a DoNotMatch error and then to a generic internal fault error with no #. So do you now want me to expose rawType on FaultEventCode and rework the ReplacePodViewController code to always display the pod reported fault # independent of whether it maps to something or not? Does the code displaying faults for an Issue Report always report the pod fault # right now and if not, should this get stuff reworked as well once the the rawType on FaultEventCode is made public?